packages/next/src/server/next-server.ts

```diff
@@ -339,6 +341,11 @@ export default class NextNodeServer extends BaseServer {
 
     // ensure options are set when loadConfig isn't called
     setHttpClientAndAgentOptions(this.nextConfig)
+
+    // Intercept fetch
+    if (this.serverOptions.experimentalTestMode) {
+      global.fetch = createTestFetch(global.fetch)
```

Comment:
In order to prevent the code for `./test` being loaded in production using `require()` is better as it prevents loading the code when the option is not set.
```suggestion
      const { createTestFetch, wrapRequestHandler } = require('./test')
      global.fetch = createTestFetch(global.fetch)
```